lib/ajax-datatables-rails/orm/active_record.rb

```diff
@@ -38,31 +38,28 @@ def composite_search(records)
         records
       end
 
-      def build_conditions_for(query)
+      def build_conditions_for(query, regex)
         search_for = query.split(' ')
         criteria = search_for.inject([]) do |criteria, atom|
-          criteria << searchable_columns.map { |col| search_condition(col, atom) }
+          criteria << searchable_columns.map { |col| search_condition(col, atom, regex == 'true') }
             .reduce(:or)
         end.reduce(:and)
         criteria
       end
 
       def aggregate_query
-        conditions = searchable_columns.each_with_index.map do |column, index|
+        conditions = view_columns.each_with_index.map do |column, index|
```

Comment:
That's actually why we need view_columns, because the `params[:columns]` array will have all the columns that are present in the view.

For example, say you have 5 columns on the view. Columns 0, 1, 2, and 4 are searchable, but 3 is not. Your `searchable_columns` will have four elements. But your `view_columns` will have 5 elements if you correctly set it up.

Now in the view you search on column 4 (the fifth one). This method would iterate over the 4 `searchable_columns` while checking the index against the 5 `params[:columns]`. It would never reach the last column, and therefore not search against it. 

So I guess we could handle this 2 ways:

1. We iterate over `view_columns` instead, and insist that the user puts all columns in that array, even if they are not searchable.

2. Perhaps better, we iterate over `searchable_columns`, but also filter out the columns from `params[:columns]` that are searchable and check for their value at that index.

Hopefully this all makes sense.